biorbd_optim/misc/optimal_control_program.py

```diff
@@ -615,6 +561,84 @@ def update_parameters(self, new_parameters):
         else:
             raise RuntimeError("new_parameter must be a ParameterOption or a ParameterList")
 
+    def update_initial_guess(self, X_init=None, U_init=None, X_bounds=None, U_bounds=None):
+        if X_bounds is not None:
+            if isinstance(X_bounds, BoundsOption):
+                X_bounds_tp = BoundsList()
+                X_bounds_tp.add(X_bounds)
+                X_bounds = X_bounds_tp
+            elif not isinstance(X_bounds, BoundsList):
+                raise RuntimeError("X_bounds should be built from a BoundOption or a BoundsList")
+            elif len(X_bounds) == 0:
+                X_bounds.add(
+                    BoundsOption(
+                        [np.zeros((self.nlp[0].model.nbQ() * 2, 1)), np.zeros((self.nlp[0].model.nbQ() * 2, 1))]
+                    )
+                )
+            self.__add_to_nlp("X_bounds", X_bounds, False)
```

Comment:
This should not be, if user sends nothing, it is is problem. A flag should be set though to remember that something is not set. This is true for X_bounds, U_bounds, X_init and U_init